beacon_node/lighthouse_network/src/peer_manager/mod.rs

```diff
@@ -1348,6 +1393,45 @@ impl<E: EthSpec> PeerManager<E> {
             }
         }
     }
+
+    fn compute_peer_custody_subnets(
+        &self,
+        peer_id: &PeerId,
+        custody_subnet_count: u64,
+    ) -> Result<HashSet<DataColumnSubnetId>, String> {
+        // If we don't have a node id, we cannot compute the custody duties anyway
+        let node_id = peer_id_to_node_id(peer_id)?;
+        let spec = &self.network_globals.spec;
+
+        if !(spec.custody_requirement..=spec.data_column_sidecar_subnet_count)
+            .contains(&custody_subnet_count)
+        {
+            return Err("Invalid custody subnet count in metadata: out of range".to_string());
+        }
+
+        let custody_subnets = DataColumnSubnetId::compute_custody_subnets::<E>(
+            node_id.raw(),
+            custody_subnet_count,
+            spec,
+        )
+        .map(|subnets| subnets.collect())
+        .unwrap_or_else(|e| {
+            // This is an unreachable scenario unless there's a bug, as we've validated the csc
+            // just above.
```

Comment:
This may be unreachable currently but there's no guarantee that it will never be.

I'd prefer to avoid `.expect` and `.unwrap` unless it's something we can  to detect on startup immediately (e.g. usage in `spec` is fine) and is not something we can recover from.

This is not a completely unrecoverable situation and we should always try to continue gracefully. We capture this event as an error anyway, so it'd still be useful for debugging without unnecessarily killing the client.